var/spack/repos/builtin/packages/autoconf/package.py

```diff
@@ -57,6 +70,10 @@ def patch(self):
                         '#! /usr/bin/env perl',
                         patched_file)
 
+    # We use autoconf's installcheck target, it runs the same testsuite as 'make check':
+    def check(self):
+        pass
```

Comment:
(question) we generally try to run both `make installcheck` and `make check`/`make test` (as part of inheriting from autotools), I'm curious if you happen to know if most packages differ (in the sense that they usually do different things when running `make check` and `make installcheck`).

Reply:
No I don't know that. AFAICS, order of the phases in `spack/build_systems/autotools.py` is:
1. def build : => make
2. def check: => if extists, make test || fail ; if exists, make check || fail
3. def install: => make install
4. def installcheck: => if exists, make installcheck

So installcheck is supposed to test if the installation works.
The installcheck of `autoconf-2.71` is different Btw: it checks if the tools can be called with --help and --version (but not more, it's just testing that "something" was installed at all, but does not exercise the scripts)

Reply:
> The installcheck of autoconf-2.71 is different Btw: it checks if the tools can be called with --help and --version (but not more, it's just testing that "something" was installed at all, but does not exercise the scripts)

In that case you could do:

```
@when('@:2.70')
def check(self):
    pass
```

In which case we will only override the `check` phase to do nothing for versions before `2.71`.